### PR TITLE
Fixes #113 - Score great grandparent nodes.

### LINKED ISSUE #113: Specific pages from ehow.com can't be displayed in Reader View
**Reproducible on**: 
Beta 38.0b1-build1 (20150330154247)

**Sample page showing this issue**:
http://www.ehow.com/how_2042752_build-terrarium.html

### PR DESCRIPTION
**Pull request by n1k0**
Spent a bunch of time investigating this one, went with scoring great grandparent nodes with good outcome — and no breakage. 

~~Note: I'm crying blood staring at the implementation, but I'm trying to focus on getting results in the shortest possible amount of time…~~

### CODE REVIEW

File: Readability.js [hunk 1]
```diff
@@ -724,8 +734,12 @@ Readability.prototype = {
         // Add the score to the parent. The grandparent gets half.
         parentNode.readability.contentScore += contentScore;
 
-        if (grandParentNode)
+        if (grandParentNode && grandParentNode.readability) {
```
Comment (gijsk) [inline Readability.js]: `_initializeNode` should be making sure that grandParentNode always exists. Does it not?

File: Readability.js [hunk 2]
```diff
@@ -724,8 +734,12 @@ Readability.prototype = {
...
           grandParentNode.readability.contentScore += contentScore / 2;
+          if (greatGrandParentNode && greatGrandParentNode.readability) {
+            greatGrandParentNode.readability.contentScore += contentScore / 4;
```
Comment (gijsk) [inline Readability.js]: Mm... I've wondered before now if we should just use some way of scoring that deals with this automagically (for whatever depth) but I've not figure out how best to do this. If this works, let's start here, I guess.

Comment (n1k0) [inline Readability.js]: Yeah, I think I can work something out here. Hold on merging :)

File: Readability.js (hunk 1 above)
Comment (n1k0) [inline Readability.js]: That's betond my current understanding, but yes, in some circumstances I didn't take the time to fully investigate, `grandParentNode` can be `undefined`.

Comment (gijsk) [inline Readability.js]: Maybe if `typeof(grandParentNode.tagName)` is undefined... that'll not initialize readability for that node...

**Comment by n1k0** [comment]
r=? @gijsk @leibovic

**Comment by gijsk** [comment]
What's the perf implication here? Assuming that's OK and with the early return restored (see comment above), r=me!

**Comment by gijsk** [comment]
(I won't be around to merge, so see above and merge when it's ready (and/or ask @leibovic ))

**Comment by n1k0** [comment]
Perf is really similar as before :)

**Pull request merged**